plugins/repository-azure/src/main/java/org/elasticsearch/repositories/azure/AzureBlobStore.java

```diff
@@ -216,91 +214,80 @@ public boolean blobExists(String blob) throws IOException {
         }
     }
 
+    // number of concurrent blob delete requests to use while bulk deleting
+    private static final int CONCURRENT_DELETES = 100;
```

Comment:
Admittedly a totally random choice of parallelism. But I figured we used up to 50 threads in the old SDK, this is certainly cheaper than additional threads -> lets at least double the concurrency.

Reply:
TBH I was expecting a higher limit :) Do you have a sense of the performance impact here?

Reply:
>  Do you have a sense of the performance impact here?

It's pretty hard to guess with the buffering overhead from Netty sending out request etc. so I figured I'd stay on the safe side. I think it's a fair guess to assume something like 2k IO buffer overhead per request and then some extra bytes for all the state handling.

> TBH I was expecting a higher limit :) 

I didn't benchmark it but I wonder how much that will even get us in terms of throughput. This isn't a batch size, this is a concurrent requests limit. So assuming 100ms latency per request this means 1k/s and it's not a global limit so concurrent delete calls would multiply this limit.
We don't really have any benchmarking here and Azure doesn't give a target request rate for deletes but their target rate for operations on a single blob is 500/s (the rate target for deletes is probably higher than this) but I figured 10x that if we run 5 snapshot threads in parallel across all blobs is a reasonable guess?